test/unit/master/test_build.py

```diff
@@ -359,23 +359,22 @@ def test_preparing_build_creates_empty_results_directory(self):
 
         self.mock_util.fs.create_dir.assert_called_once_with(build._build_results_dir())
 
-    def test_allocating_slave_to_build_sets_building_timestamp_only_on_first_slave_allocation(self):
+    def test_execute_next_subjob_or_free_executor_sets_building_timestamp_only_on_first_execution(self):
         mock_slave1 = self._create_mock_slave()
-        mock_slave2 = self._create_mock_slave()
         build = self._create_test_build(BuildStatus.PREPARED)
         scheduler = self.scheduler_pool.get(build)
+        scheduler.allocate_slave(slave=mock_slave1)
 
         self.assertIsNone(self._get_build_state_timestamp(build, BuildState.BUILDING),
                           '"building" timestamp should not be set until slave allocated.')
 
-        scheduler.allocate_slave(slave=mock_slave1)
+        scheduler.execute_next_subjob_or_free_executor(mock_slave1)
         building_timestamp1 = self._get_build_state_timestamp(build, BuildState.BUILDING)
-        scheduler.allocate_slave(slave=mock_slave2)
+        scheduler.execute_next_subjob_or_free_executor(mock_slave1)
         building_timestamp2 = self._get_build_state_timestamp(build, BuildState.BUILDING)
-
-        self.assertIsNotNone(building_timestamp1, '"building" timestamp should be set after first slave allocated.')
+        self.assertIsNotNone(building_timestamp1, '"building" timestamp should be set after first subjob is started.')
         self.assertEqual(building_timestamp1, building_timestamp2,
-                         '"building" timestamp should not change upon further slave allocation.')
+                         '"building" timestamp should not change upon further subjob execution.')
```

Comment:
Yeah, I should have changed that if I were to keep the things the way they are in the PR.